apps/asap-server/src/handlers/research-output/index-handler.ts

```diff
@@ -28,24 +28,43 @@ export const indexResearchOutputHandler = (
       SquidexWebhookResearchOutputPayload
     >,
   ): Promise<void> => {
-    const researchOutput = await researchOutputController.fetchById(
-      event.detail.payload.id,
-    );
+    logger.debug(`Event ${event['detail-type']}`);
 
-    await algoliaIndex.saveObject({
-      ...researchOutput,
-      objectID: researchOutput.id,
-    });
+    if (
+      ['ResearchOutputCreated', 'ResearchOutputUpdated'].includes(
+        event['detail-type'],
+      )
+    ) {
+      const researchOutput = await researchOutputController.fetchById(
+        event.detail.payload.id,
+      );
 
-    logger.info(`Saved Research-Output with ID ${researchOutput.id}`);
+      logger.debug(`Fetched research-output ${researchOutput.id}`);
+
+      await algoliaIndex.saveObject({
+        ...researchOutput,
+        objectID: researchOutput.id,
+      });
+
+      logger.debug(`Saved research-output ${researchOutput.id}`);
+    }
+
+    if (event['detail-type'] === 'ResearchOutputDeleted') {
```

Comment:
so there is a problem with deletes in general which i just discovered. Remember I said we should fetch the RO from squidex each time we receive this event? The reason I said it was because of the race conditions. Say somebody unpublishes a RO and then publish it again straight away - we have no guarantee that these notifications will arrive in order - either squidex or AWS may mix it up (and thats normal). If we dont check the RO state at the time of receiving the event then we may accidentally delete the RO because of the unpublish event coming after. 

So I would say that once you receive the Created/Updated event - fetch the RO from source - like you do now, which is great, but when it's deleted - you should still fetch it - if it's unpublished or deleted the call will return a 404 and thats when we know that the object should be removed, otherwise you either leave it or update it with what comes back.

What do you think?